### ISSUE TITLE
fromBase64 does not handle newlines

### ISSUE THREAD

**Issue by theduke** [unisonweb/unison#5165]
**Describe and demonstrate the bug**

The builtin `fromBase64` does not ignore newlines and instead fails with an invalid length error.

base64 allows for newlines and whitespace in the encoded content, which should be stripped out and ignored.

Newlines are pretty common in the wild. See for example how the Github API encodes base64 content: https://api.github.com/repos/unisonweb/unison/readme .

The `base64` Linux command also inserts newlines by default.

Decoding of the above works after a simple `replaceAll "\n" "" x`.

To reproduce:

```unison
  input = "IyBlYXN5ZHVyYXRpb24KClJ1c3QgbGlicmFyeSB0aGF0IHNpbXBsaWZpZXMg\nY29uc3RydWN0aW9uIG9mIGEgYHN0ZDo6dGltZTo6RHVyYXRpb25gIHdpdGgg\ndGhlCmBlYXN5ZHVyYXRpb246OkVhc3lEdXJhdGlvbmAgZXh0ZW5zaW9uIHRy\nYWl0LgoKIyMgVXNhZ2UKCkFkZCB0aGlzIHRvIHlvdXIgYENhcmdvLnRvbWxg\nOgoKYGBgdG9tbApbZGVwZW5kZW5jaWVzXQplYXN5ZHVyYXRpb24gPSAiMS4w\nLjAiCmBgYAoKYGBgcnVzdAp1c2UgZWFzeWR1cmF0aW9uOjpFYXN5RHVyYXRp\nb247CgpmbiBtYWluKCkgewogICAgbGV0IHYgPSAxLnNlY29uZHMoKTsKICAg\nIGxldCB2ID0gMTIubWludXRlcygpOwogICAgbGV0IHYgPSA1LmhvdXJzKCk7\nCn0KYGBgCgojIyBMaWNlbnNlCgpMaWNlbnNlZCB1bmRlcjoKCiogW01JVF0o\naHR0cDovL29wZW5zb3VyY2Uub3JnL2xpY2Vuc2VzL01JVCkKICBTZWUgW0xJ\nQ0VOU0VdKC4vTElDRU5TRSkK\n"

  decoded = Text.fromUtf8 (fromBase64 (Text.toUtf8 input)
```

**Comment by ceedubs** [comment]
An alternative to changing the builtin might be to change the wrapper function in the base library to do this. WDYT @runarorama @aryairani?

**Comment by aryairani** [comment]
> An alternative to changing the builtin might be to change the wrapper function in the base library to do this. WDYT @runarorama @aryairani?

Yeah, either way is probably okay. Normally we'd try to avoid changing builtins, though I doubt that anyone is relying on the "fail on newline" behavior we currently have. But it would also be quicker and less error-prone to change the wrapper function in base. We can always still update the built-in later too.
